iib/workers/tasks/legacy.py

```diff
@@ -0,0 +1,226 @@
+            raise IIBError(f'Push to {organization} in the legacy app registry was unsucessful')
+        resp.raise_for_status()
+
+
+def _get_rebuilt_index_image(request_id):
+    """
+    Generate the pull specification of the index image rebuilt by IIB.
+
+    :param int request_id: the ID of the IIB build request
+    :return: pull specification of the rebuilt index image
+    :rtype: str
+    """
+    conf = get_worker_config()
+    return conf['iib_image_push_template'].format(
+        registry=conf['iib_registry'], request_id=request_id
+    )
+
+
+def handle_legacy_content(
+    bundles, cnr_token, resolved_from_index, from_index, request_id, organization
+):
+    """
+    Handle exporting legacy content if required.
+
+    :param list bundles: a list of strings representing the bundles to be added to the index image.
+    :param str cnr_token: the token required to push back ported packages to the legacy
+        app registry via OMPS.
+    :param str resolved_from_index: resolved pull specification of the index image given by the
+        user.
+    :param str from_index: the pull specification of the index image given in the request by the
+        user.
+    :param int request_id: the ID of the IIB build request
+    :param str organization: organization name in the legacy app registry to which the back ported
+        packages should be pushed to.
+    :raises iib.exceptions.IIBError: if legacy support is required and necessary params are missing.
+    """
+    legacy_support_packages = get_legacy_support_packages(bundles)
+    if legacy_support_packages and not all([cnr_token, organization]):
+        packages_str = ', '.join(legacy_support_packages)
+        raise IIBError(
+            f'Legacy support is required for {packages_str};'
+            f' Both cnr_token and organization should be non-empty strings'
+        )
+
+    if legacy_support_packages:
+        rebuilt_index_image = _get_rebuilt_index_image(request_id)
+        opm_index_export(
+            legacy_support_packages, request_id, rebuilt_index_image, cnr_token, organization
+        )
+
+
+def opm_index_export(packages, request_id, rebuilt_index_image, cnr_token, organization):
+    """
+    Export packages to be back ported and push them via OMPS.
+
+    :param list packages: a list of strings representing the names of the packages to be exported.
+    :param int request_id: the ID of the IIB build request
+    :param str rebuilt_index_image: the pull specification of the index image rebuilt by IIB.
+    :param str cnr_token: the token required to push back ported packages to the legacy
+        app registry via OMPS.
+    :param str organization: organization name in the legacy app registry to which the back ported
+        packages should be pushed to.
+    :raises iib.exceptions.IIBError: if the export of packages fails.
+    """
+    request = get_request(request_id)
+    if request['state'] == 'failed':
```

Comment:
This is not needed since if it failed previous to this, it would have exited already.